test/uk/gov/hmrc/plasticpackagingtax/registration/controllers/amendment/AmendOrganisationDetailsControllerSpec.scala

```diff
@@ -0,0 +1,130 @@
+  private val mockAddressLookupFrontendConnector = mock[AddressLookupFrontendConnector]
+
+  private val controller =
+    new AmendOrganisationDetailsController(mockAuthAllowEnrolmentAction,
+                                           mcc,
+                                           mockAddressLookupFrontendConnector,
+                                           appConfig,
+                                           mockAmendmentJourneyAction
+    )
+
+  private val alfAddress = AddressLookupConfirmation(auditRef = "auditRef",
+                                                     id = Some("123"),
+                                                     address = AddressLookupAddress(
+                                                       lines = List("addressLine1",
+                                                                    "addressLine2",
+                                                                    "addressLine3"
+                                                       ),
+                                                       postcode = Some("E17 1ER"),
+                                                       country = Some(
+                                                         AddressLookupCountry(code = "GB",
+                                                                              name =
+                                                                                "United Kingdom"
+                                                         )
+                                                       )
+                                                     )
+  )
+
+  private def authorisedUserWithPptSubscription() =
+    authorizedUser(user =
+      newUser().copy(enrolments =
+        Enrolments(
+          Set(
+            new Enrolment(PptEnrolment.Identifier,
+                          Seq(EnrolmentIdentifier(PptEnrolment.Key, "XMPPT0000000123")),
+                          "activated"
+            )
+          )
+        )
+      )
+    )
+
+  override protected def beforeEach(): Unit = {
+    reset(mockSubscriptionConnector)
+    authorisedUserWithPptSubscription()
+    inMemoryRegistrationAmendmentRepository.reset()
+    simulateGetSubscriptionSuccess(aRegistration())
+    when(mockAddressLookupFrontendConnector.initialiseJourney(any())(any(), any())).thenReturn(
+      Future(AddressLookupOnRamp("/on-ramp"))
+    )
+    when(mockAddressLookupFrontendConnector.getAddress(any())(any(), any())).thenReturn(
+      Future.successful(alfAddress)
+    )
+  }
+
+  "Amend Organisation Details Controller" should {
+
+    "redirect to address lookup frontend" when {
+      "change business address" in {
+
+        val resp: Future[Result] = controller.changeBusinessAddress()(getRequest())
+
+        status(resp) mustBe SEE_OTHER
+        redirectLocation(resp) mustBe Some("/on-ramp")
+      }
+    }
+
+    "obtain address from address lookup and update registration and redirect to organisation list" when {
```

Comment:
We redirect to the "amend registration page" rather than the "organisation list", right?

Reply:
Also, we're not checking that the registration is updated in this test. I think we either should or update the test name.